contributing-device-support.md

```diff
@@ -0,0 +1,66 @@
+stick that is used for the initial boot. When booted, this flasher image should automatically install resin.io on internal storage
+
+    resin-image.bbappend file should define the following variables: IMAGE_FSTYPES_`<yocto-machine-name>` - this variable is used to declare the type of the produced image (it can be ext3, ext4, resin-sdcard etc. The usual type
+for a board that can boot from SD card, USB, is "resin-sdcard") and RESIN_BOOT_PARTITION_FILES_`<yocto-machine-name>` - this allows adding needed files from the deploy directory into the boot partition (can be used to add
+bootloader config files, first stage bootloader, initramfs etc. This is a list of files relative to DEPLOY_DIR_IMAGE that will be included in the vfat partition. It has the following format:
+"FilenameRelativeToDeployDir:FilenameOnTheTarget". if FilenameOnTheTarget is omitted the same filename will be used. For example to have the Intel NUC bzImage-intel-corei7-64.bin copied from deploy directory over to the
+boot partition, renamed to vmlinuz: RESIN_BOOT_PARTITION_FILES_nuc = "bzImage-intel-corei7-64.bin:vmlinuz")
+
+    resin-image-flasher.bbappend file should define the following variables: IMAGE_FSTYPES_`<yocto-machine-name>` (see above), RESIN_BOOT_PARTITION_FILES_`<yocto-machine-name>` (see above). If the board uses different bootloader
+configuration files for booting from SD/USB and internal storage, then make sure these files end up in the boot partition (i.e. they should be listed in the RESIN_BOOT_PARTITION_FILES_`<yocto-machine-name>` variable)
+
+- recipes-kernel/linux: shall contain a .bbappend to the kernel recipe used by the respective board. This kernel .bbappend must "inherit kernel-resin" in order to add the necessary kernel configs for using with resin.io
+
+- recipes-support/resin-init: shall contain a resin-init-flasher.bbappend file if you intend to install resin.io to internal storage and hence use the flasher image. This shall define the following variables:
+INTERNAL_DEVICE_KERNEL_`<yocto-machine-name>` used to identify the internal storage where resin.io will be written to; INTERNAL_DEVICE_BOOTLOADER_CONFIG_`<yocto-machine-name>` used to specify the filename of the bootloader
+configuration file the board uses when booting from internal media (must be the same with the one listed in the RESIN_BOOT_PARTITION_FILES_`<yocto-machine-name>` variable from the recipes-core/images/resin-image-flasher.bbappend file);
+INTERNAL_DEVICE_BOOTLOADER_CONFIG_PATH_`<yocto-machine-name>` used to specify the relative path (including filename) to /boot/ (this is currently partition 1) where INTERNAL_DEVICE_BOOTLOADER_CONFIG will be copied to
+(e.g.: setting INTERNAL_DEVICE_BOOTLOADER_CONFIG_beaglebone = "uEnv.txt_internal" and INTERNAL_DEVICE_BOOTLOADER_CONFIG_PATH_beaglebone = "/uEnv.txt" will end up after flashing with the file uEnv.txt_internal being copied with the
+name uEnv.txt to the root of partition 1 which is mounted at /boot/)
```

Comment:
I think I will add a section on building also. For testing shall this point to our testlodge test cases?